File: combat/src/classicthunder/combat/state/DiscardingState.java
package classicthunder.combat.state;

import classicthunder.combat.hand.Discarder;
import classicthunder.combat.hand.Hand;

public class DiscardingState implements State {

    private final Hand hand;
    private final Discarder discarder;

    public DiscardingState(Hand hand, Discarder discarder) {

        this.hand = hand;
        this.discarder = discarder;
    }

    @Override
    public void Enter() {

        hand.DiscardHand(discarder);
    }

    @Override
    public EngineState Update() {

        hand.Update();
        discarder.Update();

        if (discarder.GetCardCount() == 0) {
            return EngineState.EnemyTurn;
        }

        return EngineState.Discarding;
    }

    @Override
    public void Exit() {

    }
}

File: combat/src/classicthunder/combat/state/DiscardingState.kt
package classicthunder.combat.state

import classicthunder.combat.hand.Discarder
import classicthunder.combat.hand.Hand

internal class DiscardingState(private val hand: Hand, private val discarder: Discarder) : State {
    override fun enter() {
        hand.DiscardHand(discarder)
    }

    override fun update(): EngineState {
        hand.Update()
        discarder.update()
        return if (discarder.getCardCount() == 0) {
            EngineState.EnemyTurn
        } else EngineState.Discarding
    }

    override fun exit() {}
}